src/installer-win/SetPath.ps1

```diff
@@ -0,0 +1,49 @@
+param (
+  [Parameter()]
+  [ValidateNotNullOrEmpty()]
+  [string] $AppPath,
+
+  [Parameter()]
+  [bool]
+  $Remove = $false
+)
+
+$ErrorActionPreference = 'Stop';
+
+$registryPath = 'HKCU:\Environment';
+$environmentVariable = 'Path';
+
+$environmentKey = Get-Item -path $registryPath;
+[string]$currentPath = $environmentKey.GetValue($environmentVariable, '', [Microsoft.Win32.RegistryValueOptions]::DoNotExpandEnvironmentNames);
+
+$paths = $currentPath.Split(@(';'), [System.StringSplitOptions]::RemoveEmptyEntries);
+
+if($Remove)
+{
+  # remove path
+  if($paths -contains $AppPath)
+  {
+    $paths = $paths | Where-Object { $_ -ne $AppPath };
+    $modified = $true;
+  }
+}
+else
+{
+  # upsert path
+  if(-not ($paths -contains $AppPath))
+  {
+    $paths += $AppPath;
+    $modified = $true;
+  }
+}
+
+if(-not $paths)
+{
+  $paths = @();
+}
+
+if($modified)
+{
+  $newPath = [string]::Join(';', $paths);
+  New-ItemProperty -Path $registryPath -Name $environmentVariable -PropertyType ExpandString -Value $newPath -Force | Out-Null;
```

Comment:
Does this work without a reboot? I found that updating the reg value didn't work when I wrote the install instructions.

Reply:
Worked on my home machine immediately. The installer does something to make explorer.exe reload environment variables (there's a ChangesEnvironment=yes) setting. What you were probably seeing was explorer.exe launched processes getting stale environment.

Reply:
It will not help with any open command shells (cmd.exe, powershell, etc.) until they are reopened. However, that's a standard behavior for environments. I can make the installer force a reboot, but I don't think it's necessary.

Reply:
That's fine, no need to force a reboot. The behavior I was seeing was also with newly opened cmd/powershell windows. Sounds like the installer is doing something special to handle it.

Reply:
Yeah, explorer or windows must have some sort of API for these situations.